add extendsEnum() to support extending of enumerations

shrinktofit opened the pull request.

JoneLau commented on cocos/ui/components/button-component.ts:
```diff
@@ -79,6 +79,10 @@ enum State {
+// export enum TestButtonEventType {
+//     Click = 'click',
+// }
+
```
测试注释内容可以去除

JoneLau commented on cocos/ui/components/toggle-component.ts:
```diff
@@ -33,6 +33,11 @@ import { ccclass, executeInEditMode, executionOrder, menu, property } from '../.
+
+// enum TestToggleEventType {
+//     Toggle = 'toggle',
+// }
```
测试注释内容可以去除

pandamicro commented on cocos/core/data/utils/extends-enum.ts:
```diff
@@ -0,0 +1,46 @@
+        for (const e of enums) {
+            for (const kv of Object.keys(e)) {
+                if (kvs.indexOf(kv) >= 0) {
+                    throw new Error(`Violation error: extending enumerations shall have non-overlaped member names or member values`);
```
用 ErrorID，参考

https://github.com/cocos-creator/docs-3d/blob/master/zh/scripting/log.md

shrinktofit committed:
add extendsEnum() to support extending of enumerations


typo


update


update
Update button-component.ts
Update toggle-component.ts
Update toggle-component.ts
Update toggle-component.ts
Update button-component.ts
Update toggle-component.ts
Update button-component.ts
